buildchain/buildchain/targets/package.py

```diff
@@ -238,9 +239,13 @@ def build_srpm(self) -> types.TaskDict:
     def _download_sources(self) -> None:
         """Return a list of actions to download the source files."""
         for srcfile, url in self._get_source_files_urls().items():
-            with urllib.request.urlopen(url) as conn:
-                with open(srcfile, 'wb') as fp:
-                    fp.write(conn.read())
+            if url.startswith("http"):
```

Comment:
This is too restrictive. We could download from HTTPS or FTP.

I would write the code like that instead:

```python
            if urllib.parse.urlparse(url).scheme:
                with urllib.request.urlopen(url) as conn:
                    with open(srcfile, 'wb') as fp:
                        fp.write(conn.read())
            else:
                url = os.path.join(constants.ROOT/'packages', url)
                shutil.copyfile(url, srcfile)
```

If we have a scheme => we download, if not it's a local file and we copy it.